### PR TITLE
Raise ValueError for URL with authority with path without leading slash

### PR DESCRIPTION
**Pull request by rominf**

### CODE REVIEW

File: yarl/__init__.py [hunk 1]
```diff
@@ -612,6 +615,12 @@ def name(self):
         """The last part of parts."""
         return self._UNQUOTER(self.raw_name)
 
+    # noinspection PyPep8Naming
```
Comment (webknjaz) [inline yarl/__init__.py]: Don't do this.

File: yarl/__init__.py [hunk 2]
```diff
@@ -612,6 +615,12 @@ def name(self):
...
+    @staticmethod
+    def raise_ValueError_for_URL_with_authority_with_path_without_leading_slash(host, path):
+        if len(host) > 0 and len(path) > 0 and path[0] != '/':
```
Comment (webknjaz) [inline yarl/__init__.py]: `if len(host) > 0 and len(path) > 0 and not path.startswith('/'):`

File: yarl/__init__.py [hunk 3]
```diff
@@ -612,6 +615,12 @@ def name(self):
         """The last part of parts."""
         return self._UNQUOTER(self.raw_name)
 
+    # noinspection PyPep8Naming
+    @staticmethod
+    def raise_ValueError_for_URL_with_authority_with_path_without_leading_slash(host, path):
```
Comment (webknjaz) [inline yarl/__init__.py]: Use `validate_authority_uri_abs_path` and write a sane docstring instead.

**Comment by rominf** [comment]
Is this better?

**Comment by webknjaz** [comment]
Thanks!

**Pull request merged**